engine/gamesys/src/gamesys/components/comp_particlefx.cpp

```diff
@@ -112,9 +112,10 @@ namespace dmGameSystem
 
         dmGraphics::HVertexStreamDeclaration stream_declaration = dmGraphics::NewVertexStreamDeclaration(dmRender::GetGraphicsContext(ctx->m_RenderContext));
 
-        dmGraphics::AddVertexStream(stream_declaration, "position",  3, dmGraphics::TYPE_FLOAT, false);
-        dmGraphics::AddVertexStream(stream_declaration, "color",     4, dmGraphics::TYPE_FLOAT, true);
-        dmGraphics::AddVertexStream(stream_declaration, "texcoord0", 2, dmGraphics::TYPE_FLOAT, true);
+        dmGraphics::AddVertexStream(stream_declaration, "position",   3, dmGraphics::TYPE_FLOAT, false);
+        dmGraphics::AddVertexStream(stream_declaration, "color",      4, dmGraphics::TYPE_FLOAT, true);
+        dmGraphics::AddVertexStream(stream_declaration, "texcoord0",  2, dmGraphics::TYPE_FLOAT, true);
+        dmGraphics::AddVertexStream(stream_declaration, "page_index", 1, dmGraphics::TYPE_FLOAT, false);
```

Comment:
We can't use anything else than floats here because of old opengl versions, but we _could_ potentially skip this and offset the UV coordinates with the page index and get it by using floor(uv), except that the UV range is 0-1 inclusive, so for the corner UVs it might not work. Maybe we can compress the range to something like 0..0.9999 but that feels a bit hacky to me..